CP-54274: Add user-specific error message for vnc connection limits

stephenchengCloud opened the pull request:
Add user name and detailed reason in http response when console connection limits are exceeded.

robhoes commented on ocaml/libs/http-lib/http_svr.ml:
```diff
@@ -166,6 +166,10 @@ let response_forbidden ?req s =
 
+let response_forbidden_with_body ?req s body =
+  let version = Option.map get_return_version req in
+  response_error_html ?version s "403" "Forbidden" [] body
```
Do we want 403, not 503 as before?

minglumlu replied:
I think it is more like a service side issue. It's server that is applying the restriction.
How about 503 + a specific error message?

robhoes commented on ocaml/xapi/console.ml:
```diff
@@ -230,7 +230,19 @@ let real_proxy' ~__context ~vm vnc_port s =
+  let user = Db.Session.get_auth_user_name ~__context ~self:session_id in
+  let body =
+    Printf.sprintf
+      "<html><body><h1>Connection Limit Exceeded</h1><p>User '%s': Only 1 \
```
Why is this useful? The caller would know itself right?

lindig replied:
This could create invalid HTML if `user` has characters with special meaning in HTML that would require escaping. 

!https://miro.medium.com/v2/resize:fit:720/format:webp/1*P4nj9fJjSeJ9-c0rwSZqlg.png

stephenchengCloud commented:
Fixed the comments above and tested.
The main changes are:
1. To contain the active users in the response message, changed the `active_connections` to record the existing users and use the unique session id to identify them in case multiple connections have the same user name.
2. Made `response_custom_error` a generic functon taking error code and messages as inputs
3.  Added `escape_char` to escape html special characters.

Tested: 
(XenCenter doesn't read the http body, so only see the header in the result)
<img width="1018" height="347" alt="image" src="https://github.com/user-attachments/assets/05dd3b13-e561-4d4f-866f-ed4b3184ec61" />

stephenchengCloud commented:
> Can both changes be squashed into a single one?

Done

psafont commented:
CI is failing because a container imager can't be pulled: AWS's DNS services are down

Once it's back up again it can be rerun
